File: src/main/java/com/smartwallet/watchlist/repository/WatchlistItemRepository.java
package com.smartwallet.watchlist.repository;

import com.smartwallet.watchlist.entity.WatchlistItem;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface WatchlistItemRepository extends JpaRepository<WatchlistItem, Long> {
    List<WatchlistItem> findByWatchlistIdOrderByPositionAsc(Long watchlistId);
    Optional<WatchlistItem> findByWatchlistIdAndAssetSymbol(Long watchlistId, String assetSymbol);
    boolean existsByWatchlistIdAndAssetSymbol(Long watchlistId, String assetSymbol);
    long countByWatchlistId(Long watchlistId);
    void deleteByWatchlistId(Long watchlistId);
}

File: src/main/kotlin/com/smartwallet/watchlist/repository/WatchlistItemRepository.kt
package com.smartwallet.watchlist.repository

import com.smartwallet.watchlist.model.Watchlist
import com.smartwallet.watchlist.model.WatchlistItem
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface WatchlistItemRepository : JpaRepository<WatchlistItem, Long> {
    fun findAllByWatchlist(watchlist: Watchlist): List<WatchlistItem>
    fun findByWatchlistAndAssetSymbol(watchlist: Watchlist, assetSymbol: String): WatchlistItem?
}
